### PR TITLE
Change ManifestList to an immutable list from linked list

### PR DESCRIPTION
**Pull request by johnclara**
From https://github.com/apache/iceberg/issues/1812

Trying to match with: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseSnapshot.java#L131

Not sure if this is worth the risk of changing. A couple of our tables have some really large manifest lists and I was wondering if the micro benchmark shows anything noticable

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestRewriteManifests.java [hunk 1]
```diff
@@ -146,7 +147,7 @@ public void testReplaceManifestsSeparate() {
         .clusterBy(file -> file.path())
         .commit();
 
-    List<ManifestFile> manifests = table.currentSnapshot().allManifests();
+    List<ManifestFile> manifests = new ArrayList(table.currentSnapshot().allManifests());
```
Comment (rdblue) [inline core/src/test/java/org/apache/iceberg/TestRewriteManifests.java:150]: Why is this needed?

Comment (johnclara) [reply core/src/test/java/org/apache/iceberg/TestRewriteManifests.java:150]: @rdblue sorting isn't supported on immutable lists.

Sorry, I should've closed this earlier. I was thinking this morning that this definitely isn't worth the risk of changing

Comment (rdblue) [reply core/src/test/java/org/apache/iceberg/TestRewriteManifests.java:150]: No problem!

**Comment by johnclara** [comment]
not worth changing
